src/Sentry/PlatformAbstractions/RuntimeInfo.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using System.Runtime.InteropServices;
+
+namespace Sentry.PlatformAbstractions
+{
+    // https://github.com/dotnet/corefx/issues/17452
+    internal static class RuntimeInfo
+    {
+        private static readonly Regex RuntimeParseRegex = new Regex("^(?<name>[^\\d]*)(?<version>(\\d+\\.)+[^\\s]+)",
+            RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        /// <summary>
+        /// Gets the current runtime.
+        /// </summary>
+        /// <returns>A new instance for the current runtime</returns>
+        internal static Runtime GetRuntime()
+        {
+            var runtime = GetFromRuntimeInformation();
+            runtime ??= GetFromMonoRuntime();
+
+            runtime ??= GetFromEnvironmentVariable();
+
+#if NETFX
+            SetReleaseAndVersionNetFx(runtime);
+#elif NETSTANDARD || NETCOREAPP // Possibly .NET Core
+            SetNetCoreVersion(runtime);
+#endif
+            return runtime;
+        }
+
+        internal static Runtime? Parse(string rawRuntimeDescription, string? name = null)
+        {
+            if (rawRuntimeDescription == null)
+            {
+                return name == null
+                    ? null
+                    : new Runtime(name);
+            }
+
+            var match = RuntimeParseRegex.Match(rawRuntimeDescription);
+            if (match.Success)
+            {
+                return new Runtime(
+                    name ?? (match.Groups["name"].Value == string.Empty
+                        ? null
+                        : match.Groups["name"].Value.Trim()),
+                    match.Groups["version"].Value,
+                    raw: rawRuntimeDescription);
+            }
+
+            return new Runtime(name, raw: rawRuntimeDescription);
```

Comment:
This block and conditionals can be simplified:

```suggestion
            var match = RuntimeParseRegex.Match(rawRuntimeDescription);
            
            return new Runtime(
                name ?? (match.Groups["name"].Value == string.Empty ? null : match.Groups["name"].Value.Trim()),
                match.Groups["version"].Value == string.Empty ? null : match.Groups["version"].Value.Trim(),
                rawRuntimeDescription
            );            
```

If `match.Success == false` then `match.Groups["foo"]` always evaluates to empty string. To simplify this further, I usually have a `string.NullIfEmptyOrWhiteSpace()` extension method.